File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/inputrepresentation/InputRepresentation.java
package org.akoshterek.backgammon.agent.inputrepresentation;

import org.akoshterek.backgammon.board.Board;

/**
 * @author Alex
 *         date 12.09.2015.
 */
public interface InputRepresentation {
    String getName();

    int getRaceInputsCouns();
    int getCrashedInputsCount();
    int getContactInputsCount();

    double[] calculateRaceInputs(Board anBoard);
    double[] calculateCrashedInputs(Board anBoard);
    double[] calculateContactInputs(Board anBoard);
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/inputrepresentation/InputRepresentation.scala
package org.akoshterek.backgammon.agent.inputrepresentation

import org.akoshterek.backgammon.board.Board

trait InputRepresentation {
  val raceInputsCount: Int
  val crashedInputsCount: Int
  val contactInputsCount: Int

  def calculateRaceInputs(anBoard: Board): Array[Float]
  def calculateCrashedInputs(anBoard: Board): Array[Float]
  def calculateContactInputs(anBoard: Board): Array[Float]
}